### PR TITLE
add aks-aso chart

### PR DESCRIPTION
**Pull request by nojnhuh**

### CODE REVIEW

**Review by dtzar — CHANGES_REQUESTED** [review]
If we're going with the flat directory (no sub azure folder) how you have it, then I think we should name this chart azure-aks-aso.

**Comment by dtzar** [comment]
I also recommend putting in a basic readme at the root so people know the difference between this chart and the other CAPZ AKS chart.

**Comment by mboersma** [comment]
Apprently `helm-lint` is [overly strict](https://github.com/helm/chart-testing/issues/192) with the `maintainers:` field in Chart.yaml. It can't be empty and `name` must be a valid GitHub handle. I was investigating this in #7.

That may be fine for us, or we could consider disabling that check.

**Comment by dtzar** [comment]
No, I think we should adhere to the lint with maintainers.  It's easy to add that field in there.  The linter doesn't make you maintain the chart if you put your name in there 😅

File: charts/azure-aks-aso/templates/cluster.yaml
Comment (mboersma) [inline charts/azure-aks-aso/templates/cluster.yaml]:
```suggestion
  name: {{ include "capz.clusterName" . | quote }}
```

File: charts/azure-aks-aso/templates/cluster.yaml
Comment (mboersma) [inline charts/azure-aks-aso/templates/cluster.yaml]:
```suggestion
    name: {{ include "capz.clusterName" . | quote }}
```

File: charts/azure-aks-aso/templates/cluster.yaml
Comment (mboersma) [inline charts/azure-aks-aso/templates/cluster.yaml]:
```suggestion
    name: {{ include "capz.clusterName" . | quote }}
```

File: charts/azure-aks-aso/templates/cluster.yaml
Comment (mboersma) [inline charts/azure-aks-aso/templates/cluster.yaml]:
```suggestion
  name: {{ include "capz.clusterName" . | quote }}
```

File: charts/azure-aks-aso/templates/cluster.yaml
Comment (mboersma) [inline charts/azure-aks-aso/templates/cluster.yaml]:
```suggestion
  name: {{ include "capz.clusterName" . | quote }}
```

File: charts/azure-aks-aso/values.yaml
Comment (mboersma) [inline charts/azure-aks-aso/values.yaml]:
```suggestion
kubernetesVersion: v1.28.9
```

**Comment by nojnhuh** [comment]
Sorry, I'm not sure I'll be able to get back to this soon. @dtzar @mboersma Could you please make the follow-up changes?

**Review by dtzar — APPROVED** [review]

**Comment by mboersma** [comment]
Oops, sorry @dtzar I added a README.md after you approved this. PTAL whenever you have a chance.

**Review by dtzar — APPROVED** [review]

**Pull request merged**
